projectq/backends/_azure/_util.py

```diff
@@ -0,0 +1,324 @@
+        raise InvalidCommandError('Invalid command:', str(cmd))
+
+    gate = cmd.gate
+
+    if isinstance(gate, ControlledGate):
+        inner_gate = gate._gate  # noqa
+        while isinstance(inner_gate, ControlledGate):
+            inner_gate = inner_gate._gate  # noqa
+        gate_type = type(inner_gate)
+    elif isinstance(gate, DaggeredGate):
+        gate_type = type(gate.get_inverse())
+    else:
+        gate_type = type(gate)
+
+    gate_name = QUANTINUUM_GATE_MAP.get(gate_type)
+
+    # Daggered gates get special treatment
+    if isinstance(gate, DaggeredGate):
+        gate_name = gate_name + 'dg'
+
+    # Controlled gates get special treatment too
+    if isinstance(gate, ControlledGate):
+        all_qubits = [qb.id for qureg in cmd.qubits for qb in qureg]
+        controls = all_qubits[:gate._n]  # noqa
+        targets = all_qubits[gate._n:]  # noqa
+    else:
+        controls = [qb.id for qb in cmd.control_qubits]
+        targets = [qb.id for qureg in cmd.qubits for qb in qureg]
+
+    # Barrier gate
+    if isinstance(gate, BarrierGate):
+        qb_str = ""
+        for pos in targets:
+            qb_str += "q[{}], ".format(pos)
+        return "{} {};".format(gate_name, qb_str[:-2])
+
+    # Daggered gates
+    elif gate in (Sdag, Tdag):
+        return "{} q[{}];".format(gate_name, targets[0])
+
+    # Controlled gates
+    elif len(controls) > 0:
+        # 1-Controlled gates
+        if len(controls) == 1:
+            gate_name = 'c' + gate_name
+            return "{} q[{}], q[{}];".format(gate_name, controls[0], targets[0])
+
+        # 2-Controlled gates
+        if len(controls) == 2:
+            gate_name = 'cc' + gate_name
+            return "{} q[{}], q[{}], q[{}];".format(
+                gate_name, controls[0], controls[1], targets[0])
+
+    # Single qubit gates
+    elif len(targets) == 1:
+        # Standard gates
+        if isinstance(gate, (HGate, XGate, YGate, ZGate, SGate, TGate)):
+            return "{} q[{}];".format(gate_name, targets[0])
+
+        # Rotational gates
+        if isinstance(gate, (Rx, Ry, Rz)):
+            return "{}({}) q[{}];".format(gate_name, gate.angle, targets[0])
+
+    # Two qubit gates
+    elif len(targets) == 2:
+        # Rotational gates
+        if isinstance(gate, (Rxx, Ryy, Rzz)):
+            return "{}({}) q[{}], q[{}];".format(gate_name, gate.angle, targets[0], targets[1])
+
+    # Invalid command
+    else:
+        raise InvalidCommandError('Invalid command:', str(cmd))
+
+
+def rearrange_result(input_result, length):
+    """Turn ``input_result`` from an integer into a bit-string.
+
+    Args:
+        input_result (int): An integer representation of qubit states.
+        length (int): The total number of bits (for padding, if needed).
+
+    Returns:
+        str: A bit-string representation of ``input_result``.
+    """
+    bin_input = list(bin(input_result)[2:].rjust(length, '0'))
+    return ''.join(bin_input)[::-1]
```

Comment:
Why not use something along the lines of ` f'0b{number:0{length}b}' ?

Reply:
This `rearrange_result` is actually copy from `projectq.backends._ionq._ionq._rearrange_result`. I am thinking of importing this function from ionq backend instead of creating one.

Let me know if it sounds good.

Reply:
I just saw that it's actually duplicated in `_aqt` as well...

Could you create a new file called `_utils.py` inside `projectq/backends/` and move that function there? That would also mean deleting that definition from both:
- `projectq/backends/_ionq/_ionq_.py`
- `projectq/backends/_aqt/_aqt.py`

And adding the following to the preamble of all these files (including yours):

```python
from ._utils import _rearrange_result
```

Reply:
Update the PR with this change. Please take a look.

Reply:
Looks good to me, thanks for making those changes.